js/services/api/user-details.js

```diff
@@ -10,16 +13,54 @@ const getUserDetails = (userId) => {
     });
 };
 
-const create = (userId, data) => {
-  return api.set(`/userDetails/${userId}`, data)
-    .then(() => {
-      console.log('User details added');
+const createPotluckInvites = (userId, data) => {
+  const encodedEmail = btoa(data.emailAddress);
+
+  return firebase
+    .database()
+    .ref()
+    .child('/potLuckInvites')
+    .once('value')
+    .then((potluckIdResult) => {
+      Object.keys(potluckIdResult.val()).forEach((key) => {
+        return firebase
+          .database()
+          .ref(`/potLuckInvites/${key}/guests/${encodedEmail}`)
+          .once('value')
+          .then((emailMatch) => {
+            if (emailMatch.val()) {
+              // if any invites create potluck guest records
+              guestInvites.createPotluckGuest(key, [userId]);
+
+              potlucks.getPotluck(key)
+                .then((result) => {
+                  // create user invite records
+                  guestInvites.createUserPotluck(key, result, [userId]);
+
+                  // delete email from potluck guest email list
+                  guestInvites.deleteExistingUserGuestInvite(key, data.emailAddress);
+                })
+                .catch((error) => {
+                  console.log(error);
+                });
+
+              return true;
+            }
```

Comment:
This function got a bit scary.

The refactoring challenge is to break it into parts. I see several. There are two 'getters'. The first gets all potluck invites, the second gets a single potluck. The function to get the single potluck (`getPotluck`) is already written. Write one for getting the invites.

If you use a guard clause instead of `if (emailMatch.val())`, it'll make the flow much clearer.

You're iterating over the potluck invite list, and for some reason you're returning a value in the forEach callback function. You're also nesting promises inside the loop. Ideally promise chains should be flat. You could use promise.all again, to avoid this mix of sync and async code.

Take a crack at it! I can help you out tomorrow if you need.